src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -58,6 +59,26 @@ public void Start(ConnectionDelegate connectionDelegate)
             _ = _serviceConnectionManager.StartAsync();
         }
 
+#if NETCOREAPP3_0
+        public void Start(ConnectionDelegate connectionDelegate, Endpoint endpoint)
```

Comment:
I think we should make this logic a bit more generic and instead of passing the endpoint, pass an `Action<HttpContext>`. That will allow for future modifications without coupling this logic to adding an endpoint specifically. That way we can keep the ifdef in the AzureSignalRHostedService and all of this code can be decoupled from Endpoint.